server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -197,6 +207,12 @@ private void assertNoOutbox(CreationMessageEntry entry, MailboxSession session)
         }
     }
 
+    private void assertAtLeastOneMailbox(CreationMessageEntry entry) throws MailboxException {
+        if (entry.getValue().getMailboxIds().isEmpty()) {
```

Comment:
Why assert, throw and catch mecanism with a custom exception when we could return a boolean and use a simple if ?
Same question for existing assertIsUserOwnerOfMailboxes.
IMO, assert and error should be used for something that should NOT happen, not for checking user input.
Moreover here, this will lead to simpler code and better performance (if I am not wrong throw/catch mecanism is way more expansif than a if).

Reply:
Arg. Yet another **change the design and fix all JMAP layer** comment.

To deliver value, I'll ignore it here, but will create another PR straight away ;-)

Reply:
See https://github.com/linagora/james-project/pull/1187 and https://github.com/linagora/james-project/commit/8836ba2539017e178e2929369866b980a54c9251